subsys/mgmt/osdp/src/osdp_common.c

```diff
@@ -74,3 +79,176 @@ struct osdp_cmd *osdp_cmd_get_last(struct osdp_pd *pd)
+	/* len must be even, non-zero and len / 2 <= max_len. */
+	if ((len & 1) == 1 || len == 0 || (len / 2) > max_len) {
+		return -1;
+	}
+	len = len / 2;
+	for (i = 0; i < len; i++) {
+		nib1 = hex2int(hstr[(i * 2) + 0]);
+		nib2 = hex2int(hstr[(i * 2) + 1]);
+		if (nib1 == -1 || nib2 == -1) {
+			return -1;
+		}
+		arr[i] = nib1 << 4 | nib2;
+	}
+	return len;
+}
+
+int osdp_key_string_to_byte_arrray(uint8_t *key, const char *key_str)
+{
+	int len;
+
+	len = hstrtoa(key, 16, key_str);
+	if (len != 16) {
+		return -1;
+	}
+	return 0;
+}
+
+void osdp_encrypt(uint8_t *key, uint8_t *iv, uint8_t *data, int len)
+{
+	const struct device *dev;
+	struct cipher_ctx ctx = {
+		.keylen = 16,
+		.key.bit_stream = key,
+		.flags = CAP_NO_IV_PREFIX
+	};
+	struct cipher_pkt encrypt = {
+		.in_buf = data,
+		.in_len = len,
+		.out_buf = data,
+		.out_len = len
+	};
+
+	dev = device_get_binding(CONFIG_OSDP_CRYPTO_DRV_NAME);
+	if (dev == NULL) {
+		LOG_ERR("Failed to get crypto dev binding!");
+		return;
+	}
+
+	if (iv != NULL) {
+		if (cipher_begin_session(dev, &ctx,
+					 CRYPTO_CIPHER_ALGO_AES,
+					 CRYPTO_CIPHER_MODE_CBC,
+					 CRYPTO_CIPHER_OP_ENCRYPT)) {
+			LOG_ERR("Failed at cipher_begin_session");
+			return;
+		}
+		if (cipher_cbc_op(&ctx, &encrypt, iv)) {
+			LOG_ERR("CBC ENCRYPT - Failed");
+		}
+	} else {
+		if (cipher_begin_session(dev, &ctx,
+					 CRYPTO_CIPHER_ALGO_AES,
+					 CRYPTO_CIPHER_MODE_ECB,
+					 CRYPTO_CIPHER_OP_ENCRYPT)) {
+			LOG_ERR("Failed at cipher_begin_session");
+			return;
+		}
+		if (cipher_block_op(&ctx, &encrypt)) {
+			LOG_ERR("ECB ENCRYPT - Failed");
+		}
+	}
+	cipher_free_session(dev, &ctx);
+}
+
+void osdp_decrypt(uint8_t *key, uint8_t *iv, uint8_t *data, int len)
+{
+	const struct device *dev;
+	struct cipher_ctx ctx = {
+		.keylen = 16,
+		.key.bit_stream = key,
+		.flags = CAP_NO_IV_PREFIX
+	};
+	struct cipher_pkt decrypt = {
+		.in_buf = data,
+		.in_len = len,
+		.out_buf = data,
+		.out_len = len
+	};
+
+	dev = device_get_binding(CONFIG_OSDP_CRYPTO_DRV_NAME);
+	if (dev == NULL) {
+		LOG_ERR("Failed to get crypto dev binding!");
+		return;
+	}
+
+	if (iv != NULL) {
+		if (cipher_begin_session(dev, &ctx,
+					 CRYPTO_CIPHER_ALGO_AES,
+					 CRYPTO_CIPHER_MODE_CBC,
+					 CRYPTO_CIPHER_OP_DECRYPT)) {
+			LOG_ERR("Failed at cipher_begin_session");
+			return;
+		}
+		if (cipher_cbc_op(&ctx, &decrypt, iv)) {
+			LOG_ERR("CBC DECRYPT - Failed");
+		}
+	} else {
+		if (cipher_begin_session(dev, &ctx,
+					 CRYPTO_CIPHER_ALGO_AES,
+					 CRYPTO_CIPHER_MODE_ECB,
+					 CRYPTO_CIPHER_OP_DECRYPT)) {
+			LOG_ERR("Failed at cipher_begin_session");
+			return;
+		}
+		if (cipher_block_op(&ctx, &decrypt)) {
+			LOG_ERR("ECB DECRYPT - Failed");
+		}
+	}
+	cipher_free_session(dev, &ctx);
+}
+
+void osdp_fill_random(uint8_t *buf, int len)
+{
+	sys_rand_get(buf, len);
+}
+
+uint32_t osdp_get_sc_status_mask(void)
+{
+	uint32_t mask = 0;
+	struct osdp_pd *pd;
+	struct osdp *ctx = osdp_get_ctx();
+
+#ifdef CONFIG_OSDP_MODE_CP
```

Comment:
Better would be to use `if (IS_ENABLED(....)) {` here if possible

Reply:
The condition check was unnecessary. Got rid of it.